angular-ngrx-scss/src/app/shared/components/repo-controls/repo-controls.component.ts

```diff
@@ -1,11 +1,43 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { FormControl } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import { debounceTime, distinctUntilChanged, combineLatest } from 'rxjs';
+import { startWith } from 'rxjs/operators';
+import { setSortAndFilterProperties } from 'src/app/state/profile/profile.actions';
+import {
+  hasActiveSortAndFilters,
+  selectFilterBySearch,
+  selectReposCount,
+} from 'src/app/state/profile/profile.selectors';
+import { SortAndFilterState } from 'src/app/state/profile/profile.state';
 
 @Component({
   selector: 'app-repo-controls',
   templateUrl: './repo-controls.component.html',
   styleUrls: ['./repo-controls.component.scss'],
 })
-export class RepoControlsComponent {
+export class RepoControlsComponent implements OnInit {
   searchInput = new FormControl('');
+  hasActiveSortAndFilters$ = this.store.select(hasActiveSortAndFilters);
+  selectReposCount$ = this.store.select(selectReposCount);
+
+  selectFilterBySearch$ = this.store.select(selectFilterBySearch);
+  constructor(private store: Store) {}
```

Comment:
nitpick(style): Keep spacing between class field definitions and function definitions consistent.

```suggestion
  searchInput = new FormControl('');
  hasActiveSortAndFilters$ = this.store.select(hasActiveSortAndFilters);
  selectReposCount$ = this.store.select(selectReposCount);
  selectFilterBySearch$ = this.store.select(selectFilterBySearch);

  constructor(private store: Store) {}
```